modules/core/src/matrix.cpp

```
@@ -1261,6 +1261,16 @@ Mat Mat::reshape(int _cn, const std::vector<int>& _newshape) const
     return reshape(_cn, (int)_newshape.size(), &_newshape[0]);
 }
 
+Mat Mat::reinterpret(int type) const
+{
+    type = CV_MAT_TYPE(type);
```

Why do we need `CV_MAT_TYPE` operation?

---

I want to retrieve the actual type of the cv::Mat to prevent users from passing incorrect parameters, and then perform an assertion check.

---

Need to start with "bad param" test then.
Silent ignoring of invalid input doesn't look good.

---

I had added assert checks `CV_Assert(CV_ELEM_SIZE(this->type()) == CV_ELEM_SIZE(type));` 

Surely, I will add test cases, and should I create a new PR to update this issue?

---

It is still not clear why do we need `type = CV_MAT_TYPE(type)` operation:
- it is useless for valid MatType (works as no-op).
- it silently ignores invalid out-of-range input converting/truncating it to some value (which user may not expect).

---

My initial thought was that types with the same element size could perform a reinterpret operation, and for any other type, an assert operation would be executed. Your point seems to be:

When newtype < this->type, a reinterpret operation (truncate) can also be performed.
When newtype > this->type, a new mat (convert) should be created.

The CV_MAT_TYPE operation is indeed redundant, as these operations are actually the responsibility of the convertTo function. CV_MAT_TYPE is indeed unnecessary.